textattack/search_methods/PSO_algorithm.py

```diff
@@ -0,0 +1,266 @@
+"""
+Reimplementation of search method from Word-level Textual Adversarial Attacking as Combinatorial Optimization
+by Zang et. al
+`<https://www.aclweb.org/anthology/2020.acl-main.540.pdf>`_
+`<https://github.com/thunlp/SememePSO-Attack>`_
+"""
+
+from copy import deepcopy
+
+import numpy as np
+
+from textattack.search_methods import SearchMethod
+
+
+class PSOAlgorithm(SearchMethod):
+    """
+    Attacks a model with word substiutitions using a PSO algorithm.
```

Comment:
can you say "using a Particle Swarm Optimization algorithm" here, since some people might stumble upon this file and not know what PSO stands for.